server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultCriteria.java

```diff
@@ -0,0 +1,119 @@
+/*-------------------------------------------------------------------------------------------------------------------*\
+|  Copyright (C) 2015 PayPal                                                                                          |
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets.transfer;
+
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.paypal.selion.grid.servlets.transfer.UploadRequestProcessor.RequestHeaders;
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * {@link Criteria} to match a {@link DefaultManagedArtifact} uniquely. Criteria uses artifact name, user id and
+ * application folder to uniquely identify a {@link DefaultManagedArtifact}. Parameters artifactName, userId and
+ * applicationFolder match artifact name, folder name and parent folder name of some {@link DefaultManagedArtifact}
+ * respectively.
+ */
+public final class DefaultCriteria implements Criteria {
+    
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(DefaultCriteria.class);
+
+    private String artifactName;
+
+    private String userId;
+
+    private String applicationFolder;
+
+    public DefaultCriteria(EnumMap<RequestHeaders, String> parametersMap) {
+        validateParametersMap(parametersMap);
+        this.artifactName = parametersMap.get(RequestHeaders.FILENAME);
+        this.userId = parametersMap.get(RequestHeaders.USERID);
```

Comment:
I agree, all upload side is limited by headers. I see how it defeats the purpose of the changes I submitted. Maybe, I should have modified `UploadedArtifact` to contain all header name-values in a property, and let `ServerRepository#saveContents` use it.